graylog2-server/src/main/java/org/graylog2/indexer/indexset/IndexSetConfig.java

```diff
@@ -91,7 +89,7 @@ public static IndexSetConfig create(@Id @ObjectId @JsonProperty("_id") @Nullable
                                         @JsonProperty("index_optimization_disabled") @Nullable Boolean indexOptimizationDisabled,
                                         @JsonProperty("field_type_refresh_interval") @Nullable Duration fieldTypeRefreshInterval,
                                         @JsonProperty("custom_field_mappings") @Nullable CustomFieldMappings customFieldMappings,
-                                        @JsonProperty(FIELD_PROFILE_ID) @Nullable String fieldTypeProfile,
+                                        @JsonProperty(FIELD_PROFILE_ID) @Pattern(regexp = "^[a-fA-F\\d]{24}$") @Nullable String fieldTypeProfile,
```

Comment:
What about creating a `@ValidObjectId` annotation with a validator that is using `ObjectId.isValid`?

Reply:
It will require some additional code, but for potential future use-cases may be useful. I'll try to do it.